ui/app/components/mount-accessor-select.js

```diff
@@ -1,34 +1,49 @@
+import Component from '@glimmer/component';
 import { inject as service } from '@ember/service';
-import Component from '@ember/component';
-import { task } from 'ember-concurrency';
+import { dropTask } from 'ember-concurrency';
+import { action } from '@ember/object';
 
-export default Component.extend({
-  store: service(),
+/**
+ * @module MountAccessorSelect
+ * The MountAccessorSelect component is used to selectDrop down mount options.
+ *
+ * @example
+ * ```js
+ * <MountAccessorSelect @value={this.aliasMountAccessor} @onChange={this.onChange} />
+ * ```
+ * @param {string} value - the selected value.
+ * @param {function} onChange - the parent function that handles when a new value is selected.
+ * @param {boolean} [showAccessor] - whether or not you should show the value or the more detailed accessor off the class.
+ * @param {boolean} [noDefault] - whether or not there is a default value.
+ * @param {boolean} [filterToken] - whether or not you should filter out type "token".
+ * @param {string} [name] - name on the label.
+ * @param {string} [label] - label above the select input.
+ * @param {string} [helpText] - text shown in tooltip.
+ */
 
-  // Public API
-  //value for the external mount selector
-  value: null,
-  filterToken: false,
-  noDefault: false,
-  onChange: () => {},
+export default class MountAccessorSelect extends Component {
+  @service store;
 
-  init() {
-    this._super(...arguments);
+  filterToken = false;
+  noDefault = false;
+  value = '';
+
+  constructor() {
+    super(...arguments);
     this.authMethods.perform();
-  },
+  }
 
-  authMethods: task(function* () {
+  @dropTask *authMethods() {
     let methods = yield this.store.findAll('auth-method');
-    if (!this.value && !this.noDefault) {
-      this.set('value', methods.get('firstObject.accessor'));
-      this.onChange(this.value);
+    if (!this.args.value && !this.args.noDefault) {
+      // ARG can't set value as args.
```

Comment:
This comment is a tad unclear, not sure if it's a leftover note-to-self or to help explain the following lines.